npc/controller.go

```diff
@@ -121,11 +124,24 @@ func (npc *controller) AddNetworkPolicy(obj interface{}) error {
 	npc.Lock()
 	defer npc.Unlock()
 
+	// lazily add default rule to drop egress traffic only when network policies are applied
+	if !npc.defaultEgressDrop {
+		npc.defaultEgressDrop = true
+		ipt, err := goiptables.New()
+		if err != nil {
+			common.Log.Fatalf("Failed to create iptables handler %v", err.Error())
+		}
+		if err := ipt.Append(TableFilter, EgressChain,
+			"-m", "mark", "!", "--mark", EgressMark, "-j", "DROP"); err != nil {
+			common.Log.Errorf("Failed to add iptable rule to drop egress traffic from the pods by default due to %s", err.Error())
+			npc.defaultEgressDrop = false
```

Comment:
Is there any point in continuing here?  This policy isn't going to achieve much if we don't drop packets.

Reply:
Agree. Since this base rule is important only for egress network policies. I made change to restrict it only for egress network policies.